typegen/unification/subtyping.py

```diff
@@ -0,0 +1,209 @@
+import importlib
+import importlib.util
+from importlib.machinery import SourceFileLoader
+import logging
+import os
+from types import ModuleType
+import pandas as pd
+import pathlib
+import sys
+
+from .filter_base import TraceDataFilter
+import constants
+
+logger = logging.getLogger(__name__)
+
+
+def _none_if_builtin(module: str) -> str | None:
+    return module if module != "builtins" else None
+
+
+def _attempt_module_lookup(
+    module_name: str, root: pathlib.Path, lookup_path: pathlib.Path
+) -> ModuleType | None:
+    try:
+        loader = SourceFileLoader(module_name, str(root / lookup_path))
+        spec = importlib.util.spec_from_loader(loader.name, loader)
+        if spec is not None:
+            module = importlib.util.module_from_spec(spec)
+
+            sys.modules[module_name] = module
+            loader.exec_module(module)
+            del sys.modules[module_name]
+
+            logger.debug(f"Imported {module_name} from {str(root / lookup_path)}")
+            return module
+
+    except FileNotFoundError:
+        logger.debug(f"Could not import {module_name} from {str(root / lookup_path)}")
+
+    return None
+
+
+class ReplaceSubTypesFilter(TraceDataFilter):
+    """Replaces rows containing types in the data with their common base type."""
+
+    ident = "repl_subty"
+
+    stdlib_path: pathlib.Path | None = None
```

Comment:
As mentioned before, I recommend moving the imports to another class. Add an instance to the filter.